xiangqi_viewer.js

```diff
@@ -106,25 +106,34 @@ XiangqiViewer.BoardRenderer = function(selector, cellSize, strokeWidth) {
   }
 
   this.movePiece = function(file, rank, piece) {
-    piece.rendered.attr({x: getX(file), y: getY(rank)});
+    piece.rendered.animate({x: getX(file), y: getY(rank)}, 1000, mina.easeinout);
   };
 
   this.highlight = function(position) {
     var x = getX(position.file) + (PIECE_SIZE / 2);
     var y = getY(position.rank) + (PIECE_SIZE / 2);
 
-    highlighted.push(root.circle(x, y, PIECE_SIZE * 1.2 / 2)
+    highlighted.push(root.circle(x, y, PIECE_SIZE * 1.2 / 2).prependTo(root)
```

Comment:
> why do we need to prepend to root here?

For layering so that the pieces move on top of the green stroke. Unfortunately with SVG's you can't use a property like `z-index` so you accomplish this by the ordering of the elements in the document.